ethcore/src/client/client.rs

```diff
@@ -417,21 +418,41 @@ impl<V> BlockChainClient for Client<V> where V: Verifier {
 
 		// Add transactions
 		let block_number = b.block().header().number();
+		let gas_limit = *b.block().header().gas_limit();
+		let mut gas_left = gas_limit;
+		let mut invalid_transactions = HashSet::new();
+
 		for tx in transactions {
```

Comment:
Stop when gas_left < min transaction gas (21000). There should be a constant for it in the Schedule I believe